app/services/pending_payments.rb

```diff
@@ -6,7 +6,7 @@ def initialize(order)
   end
 
   def payment_object
-    @order.payments.select{ |p| p if p.state == 'checkout' }.first
+    @order.payments.select{ |payment| payment if payment.state == 'checkout' }.first
```

Comment:
I can find another thing to pick on. ;-) The `select` block is evaluated as boolean, so you can scrap the `payment if`:
```rb
@order.payments.select{ |payment| payment.state == 'checkout' }.first
```

But you actually don't need to iterate all payments just to select the first one in checkout. There is a better method:
```ruby
@order.payments.find{ |payment| payment.state == 'checkout' }
```

It's not useful in this case, but you could use lazy evaluation as well:
```rb
@order.payments.lazy.select{ |payment| payment.state == 'checkout' }.first
```

Lazy evaluation helps with big enumerations. You can even deal with infinity:
```ruby
(1..Float::INFINITY).lazy.select { |i| i == 5 }.first
# 5
```